src/libpgmoneta/value.c

```diff
@@ -456,14 +457,24 @@ string_to_string_cb(uintptr_t data, int32_t format, char* tag, int indent)
    {
       if (format == FORMAT_JSON || format == FORMAT_JSON_COMPACT)
       {
-         snprintf(buf, MISC_LENGTH, "\"%s\"", str);
+         for (int i = 0; i < strlen(str); i++)
+         {
+            if (str[i] == '\'' || str[i] == '\"')
+            {
+               translated_ec_string = pgmoneta_append_char(translated_ec_string, '\\');
+            }
+            translated_ec_string = pgmoneta_append_char(translated_ec_string, str[i]);
```

Comment:
Append is kinda expensive, perhaps do a scan first to decide if you need translation, otherwise just do snprintf.

Reply:
I would prefer just calculate how much space I need, malloc the memory and copy the characters in place one by one. Then the whole thing would be O(n).